### PR TITLE
Change VB compiler switch from \ to -

### LINKED ISSUE #4450: VB compiler switch usage
Recently we found out that our [C# compiler options articles](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-options/listed-alphabetically) were not using a consistent switch for those pages. Sometimes it would use - and sometimes it would use /, since both are accepted. 

We asked the C# community and the overwhelming majority voted in issue #4072 to use - for the compiler switch. So in C# topics and multi-language topics we're adopting - for the switch.

Now, we want to ask the VB community what is your preference. I personally don't have a preference for this one.

Respond 👍 to keep `/` for compiler switches, 👎 if you'd prefer to change it to '-'.

/cc @KathleenDollard

### PR DESCRIPTION
**Pull request by rpetrusha**
# Change VB compiler switch from \ to -

This PR changes VB compiler syntax to use `-` rather than `/` as the compiler switch. I've also changed some instances of C# compiler syntax, and I've corrected some miscellaneous errors in the documentation as well.

Fixes #4450

## Suggested Reviewers

@BillWagner

### CODE REVIEW

**Comment by BillWagner** [comment]
I have this a quick look, but not a super thorough read. Let me know when it's ready for a full review.

One comment is that we should add a statement that both `-` and `/` are supported [similar to this statement in the C# docs](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-options/)

Should that also be an include file so it appears on each option page? If you think so, I'll update the C# ones.

File: docs/framework/data/adonet/dataset-datatable-dataview/consuming-a-dataset-from-an-xml-web-service.md [hunk 1]
```diff
@@ -26,7 +26,7 @@ The <xref:System.Data.DataSet> was architected with a disconnected design, in pa
  The following example shows how to create an XML Web service and client that use the **DataSet** to transport relational data (including modified data) and resolve any updates back to the original data source.  
   
 > [!NOTE]
->  We recommend that you always consider security implications when creating an XML Web service. For information on securing an XML Web service, see [Securing XML Web Services Created Using ASP.NET](http://msdn.microsoft.com/library/354b2ab1-2782-4542-b32a-dc560178b90c).  
+>  We recommend that you always consider security implications when creating an XML Web service. For information on securing an XML Web service, see [Securing XML Web Services Created Using ASP.NET](http://msdn.microsoft.com-library/354b2ab1-2782-4542-b32a-dc560178b90c).
```
Comment (BillWagner) [inline docs/framework/data/adonet/dataset-datatable-dataview/consuming-a-dataset-from-an-xml-web-service.md]: "msdn.microsoft.com-library" should not have changed from "msdn.microsoft.com/library"

File: docs/framework/data/adonet/dataset-datatable-dataview/consuming-a-dataset-from-an-xml-web-service.md [hunk 2]
```diff
@@ -274,5 +274,5 @@ The <xref:System.Data.DataSet> was architected with a disconnected design, in pa
  [Populating a DataSet from a DataAdapter](../../../../../docs/framework/data/adonet/populating-a-dataset-from-a-dataadapter.md)  
  [Updating Data Sources with DataAdapters](../../../../../docs/framework/data/adonet/updating-data-sources-with-dataadapters.md)  
  [DataAdapter Parameters](../../../../../docs/framework/data/adonet/dataadapter-parameters.md)  
- [Web Services Description Language Tool (Wsdl.exe)](http://msdn.microsoft.com/library/b9210348-8bc2-4367-8c91-d1a04b403e88)  
+ [Web Services Description Language Tool (Wsdl.exe)](http://msdn.microsoft.com-library/b9210348-8bc2-4367-8c91-d1a04b403e88)
```
Comment (BillWagner) [inline docs/framework/data/adonet/dataset-datatable-dataview/consuming-a-dataset-from-an-xml-web-service.md]: another path change ".com-library" => ".com/library"

File: docs/framework/winforms/how-to-create-a-windows-forms-application-from-the-command-line.md [hunk 3]
```diff
@@ -46,9 +44,9 @@ The following procedures describe the basic steps that you must complete to crea
   
 4.  Add a `Main` method to the class.  
   
-    1.  Apply the <xref:System.STAThreadAttribute> to the `Main` method to specify your Windows Forms application is a single threaded apartment.  
+    1.  Apply the <xref:System.STAThreadAttribute> to the C# `Main` method to specify your Windows Forms application is a single-threaded apartment. (The attribute is not necessary in Visual Basic, since Windows forms applications developed with Visual Basic use a single-threaded apartment model by default.)  
   
-    2.  Call <xref:System.Windows.Forms.Application.EnableVisualStyles%2A> to give a Windows XP appearance to your application.  
+    2.  Call <xref:System.Windows.Forms.Application.EnableVisualStyles%2A> to apply operating system styles to your application.
```
Comment (BillWagner) [inline docs/framework/winforms/how-to-create-a-windows-forms-application-from-the-command-line.md]: I laughed here. Good change.

File: docs/visual-basic/reference/command-line-compiler/moduleassemblyname.md [hunk 4]
```diff
@@ -32,9 +30,9 @@ Specifies the name of the assembly that this module will be a part of.
 |`assembly_name`|The name of the assembly that this module will be a part of.|  
   
 ## Remarks  
- The compiler processes the `/moduleassemblyname` option only if the `/target:module` option has been specified. This causes the compiler to create a module. The module created by the compiler is valid only for the assembly specified with the `/moduleassemblyname` option. If you place the module in a different assembly, run-time errors will occur.  
+ The compiler processes the `-modulleassemblyname` option only if the `-target:module` option has been specified. This causes the compiler to create a module. The module created by the compiler is valid only for the assembly specified with the `-modulleassemblyname` option. If you place the module in a different assembly, run-time errors will occur.
```
Comment (BillWagner) [inline docs/visual-basic/reference/command-line-compiler/moduleassemblyname.md]: typo: modulleassemblyname => moduleassemblyname

File: docs/visual-basic/reference/command-line-compiler/moduleassemblyname.md [hunk 5]
```diff
@@ -32,9 +30,9 @@ Specifies the name of the assembly that this module will be a part of.
 |`assembly_name`|The name of the assembly that this module will be a part of.|  
   
 ## Remarks  
- The compiler processes the `/moduleassemblyname` option only if the `/target:module` option has been specified. This causes the compiler to create a module. The module created by the compiler is valid only for the assembly specified with the `/moduleassemblyname` option. If you place the module in a different assembly, run-time errors will occur.  
+ The compiler processes the `-modulleassemblyname` option only if the `-target:module` option has been specified. This causes the compiler to create a module. The module created by the compiler is valid only for the assembly specified with the `-modulleassemblyname` option. If you place the module in a different assembly, run-time errors will occur.  
   
- The `/moduleassemblyname` option is needed only when the following are true:  
+ The `-modulleassemblyname` option is needed only when the following are true:
```
Comment (BillWagner) [inline docs/visual-basic/reference/command-line-compiler/moduleassemblyname.md]: modulleassemblyname

File: docs/visual-basic/reference/command-line-compiler/moduleassemblyname.md [hunk 6]
```diff
@@ -43,15 +41,15 @@ Specifies the name of the assembly that this module will be a part of.
  For more information about creating a module, see [/target (Visual Basic)](../../../visual-basic/reference/command-line-compiler/target.md). For more information about friend assemblies, see [Friend Assemblies](http://msdn.microsoft.com/library/df0c70ea-2c2a-4bdc-9526-df951ad2d055).  
   
 > [!NOTE]
->  The `/moduleassemblyname` option is not available from within the Visual Studio development environment; it is available only when you compile from a command prompt.  
+>  The `-modulleassemblyname` option is not available from within the Visual Studio development environment; it is available only when you compile from a command prompt.
```
Comment (BillWagner) [inline docs/visual-basic/reference/command-line-compiler/moduleassemblyname.md]: modulleassemblyname

**Review by BillWagner — APPROVED** [review]
A very thorough update @rpetrusha Thanks for all the work.

I'm going to approve this, but you should fix the URL and other typos I found. After that, :shipit: !

**Comment by rpetrusha** [comment]
@BillWagner, I've addressed the immediate issues you found, so I'll merge this after the build completes. But there are two issues, one that you raised and the other that I discovered:
1. Whether we should use an includes to note in each compiler switch topic that both slash and hyphen are supported. We definitely should; I'll open an issue to create the include. When it's created, we can include the compiler topic.
2. I forgot to search for and therefore didn't change occurrences of "vbc", "csc", "vbc.exe", or "csc.exe' in source code. There are some, but since comments aren't localized, we should open an issue to move these blocks of text from comments into the example descriptions.

**Pull request merged**